test/test_iterstream/test_stream.py

```diff
@@ -58,6 +58,33 @@ def test_take(samples: t.List[SampleType]) -> None:
     assert len(res) == len(samples)
 
 
+def test_take_exact(samples: t.List[SampleType], num_samples: int) -> None:
+    """Test take_exact"""
+
+    # take all elements in iterator
+    it = IterableSource(samples).loop().take(len(samples)).collect()
+    assert len(list(it)) == len(samples)
```

Comment:
Do we need the list-casting here (since we are collecting the iterable in the line above)? If no, we can remove it. There are a few other ones below as well
```suggestion
    assert len(it) == len(samples)
```